group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,574 @@
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    @Test
+    public void testFirstAssignmentNumConsumersGreaterThanNumPartitions() {
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(3));
+        topics.put(topic3Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        // Initial Subscriptions: A -> T1, T3 | B -> T1, T3 | C -> T1, T3
+
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        members.put(consumerC, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 3 has 2 partitions but three consumers subscribed to it - one of them will not get a partition.
+        // Topic 1 Partitions Assignment
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(1));
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(2));
+        // Topic 2 Partitions Assignment
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(1));
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    @Test
+    public void testReassignmentNumConsumersGreaterThanNumPartitionsWhenOneConsumerAdded() {
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(2));
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        // Initial Subscriptions: A -> T1, T2 | B -> T1, T2 | C -> T1, T2
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, Collections.singleton(0));
+        currentAssignmentForA.put(topic2Uuid, Collections.singleton(0));
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForA)
```

Comment:
nit: Here we could use my `mkAssignment` helper method and inline the current assignment. The would reduce the boilerplate.

Reply:
I think for readability its fine to have currentAssignment for B and then just pass it